internal/agent/device/spec/manager.go

```diff
@@ -0,0 +1,524 @@
+package spec
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/agent/client"
+	"github.com/flightctl/flightctl/internal/agent/device/errors"
+	"github.com/flightctl/flightctl/internal/agent/device/fileio"
+	"github.com/flightctl/flightctl/internal/container"
+	"github.com/flightctl/flightctl/pkg/log"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ Manager = (*manager)(nil)
```

Comment:
Can I get a brief explanation of what this `var _ Manager = (*manager)(nil)` syntax is used for?

Reply:
Hey great question this syntax is an assertion (guarantee) that at compile time the object `*manager` implements the interface `Manager`. It's sort of a cheap safety net.

Reply:
To see how it works adjust the signature of one of the methods in the implementation and try to compile.

Reply:
Ok cool, thanks for the info!!